R/teal_data-datanames.R

```diff
@@ -43,10 +43,25 @@ setGeneric("datanames<-", function(x, value) standardGeneric("datanames<-"))
 setMethod("datanames<-", signature = c("teal_data", "character"), definition = function(x, value) {
   checkmate::assert_subset(value, names(x@env))
   x@datanames <- value
+  x@datanames <- sort_datanames(x@datanames, x@join_keys)
   methods::validObject(x)
   x
 })
 setMethod("datanames<-", signature = c("qenv.error", "character"), definition = function(x, value) {
   methods::validObject(x)
   x
 })
+
+
+#' @keywords internal
+sort_datanames <- function(datanames, joinkeys) {
+
+  child_parent <- sapply(
+    datanames,
+    function(name) teal.data::parent(joinkeys, name),
```

Comment:
The prefix can be dropped now that we're in `teal.data`

```suggestion
    function(name) parent(joinkeys, name),
```